config/index.ts

```diff
@@ -1,268 +1,406 @@
-}
 
-export function updateDefaultAccount(nameOrId: string | number): void {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.updateDefaultAccount(nameOrId);
-  } else {
-    config_DEPRECATED.updateDefaultAccount(nameOrId);
+  const account = getConfigAccountByIdentifier(
+    accounts,
+    'accountId',
+    defaultAccount
+  );
+
+  if (!account) {
+    throw new Error(
+      i18n('config.getConfigDefaultAccount.accountMissingError', {
+        defaultAccount,
+      })
+    );
   }
+
+  return account;
 }
 
-export async function renameAccount(
-  currentName: string,
-  newName: string
-): Promise<void> {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.renameAccount(currentName, newName);
-  } else {
-    config_DEPRECATED.renameAccount(currentName, newName);
-  }
+export function getAllConfigAccounts(): HubSpotConfigAccount[] {
+  const { accounts } = getConfig();
+
+  return accounts;
 }
 
-export function getAccountId(nameOrId?: string | number): number | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.getAccountId(nameOrId);
+export function getConfigAccountEnvironment(
+  identifier?: number | string
+): Environment {
+  if (identifier) {
+    const config = getConfig();
+
+    const account = getConfigAccountByInferredIdentifier(
+      config.accounts,
+      identifier
+    );
+
+    if (account) {
+      return account.env;
+    }
   }
-  return config_DEPRECATED.getAccountId(nameOrId) || null;
+  const defaultAccount = getConfigDefaultAccount();
+  return defaultAccount.env;
 }
 
-export function removeSandboxAccountFromConfig(
-  nameOrId: string | number
-): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.removeAccountFromConfig(nameOrId);
+export function addConfigAccount(accountToAdd: HubSpotConfigAccount): void {
+  if (!isConfigAccountValid(accountToAdd)) {
+    throw new Error(i18n('config.addConfigAccount.invalidAccount'));
   }
-  return config_DEPRECATED.removeSandboxAccountFromConfig(nameOrId);
-}
 
-export async function deleteAccount(accountName: string): Promise<void> {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.removeAccountFromConfig(accountName);
-  } else {
-    config_DEPRECATED.deleteAccount(accountName);
+  const config = getConfig();
+
+  const accountInConfig = getConfigAccountByIdentifier(
+    config.accounts,
+    'accountId',
+    accountToAdd.accountId
+  );
+
+  if (accountInConfig) {
+    throw new Error(
+      i18n('config.addConfigAccount.duplicateAccount', {
+        accountId: accountToAdd.accountId,
+      })
+    );
   }
+
+  config.accounts.push(accountToAdd);
+
+  writeConfigFile(config, getConfigFilePath());
 }
 
-export function updateHttpTimeout(timeout: string): void {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.updateHttpTimeout(timeout);
-  } else {
-    config_DEPRECATED.updateHttpTimeout(timeout);
+export function updateConfigAccount(
+  updatedAccount: HubSpotConfigAccount
```

Comment:
This function now accepts a full updated account rather than just fields to update. This makes it easier to use TS to confirm that updates are valid (for example, TS would have no way to know the auth type of the account you're trying to update, and therefore would allow updating accounts to have invalid auth)